sdk/tools/azure-openrewrite/pom.xml

```diff
@@ -31,14 +35,31 @@
             <scope>compile</scope>
         </dependency>
 
-        <!-- You only need the version that corresponds to your current
-        Java version. It is fine to add all of them, though, as
-        they can coexist on a classpath. -->
-        <dependency>
-            <groupId>org.openrewrite</groupId>
-            <artifactId>rewrite-java-8</artifactId>
-            <scope>test</scope>
-        </dependency>
+      <!-- To test on different JDKs multiple within CI,
+      we need to include multiple rewrite-java test dependencies -->
+      <dependency>
+          <groupId>org.openrewrite</groupId>
+          <artifactId>rewrite-java-8</artifactId>
+          <scope>test</scope>
+      </dependency>
+
+      <dependency>
+        <groupId>org.openrewrite</groupId>
+        <artifactId>rewrite-java-11</artifactId>
+        <scope>test</scope>
+      </dependency>
+
+      <dependency>
+        <groupId>org.openrewrite</groupId>
+        <artifactId>rewrite-java-17</artifactId>
+        <scope>test</scope>
+      </dependency>
+
+      <dependency>
+        <groupId>org.openrewrite</groupId>
+        <artifactId>rewrite-java-21</artifactId>
+        <scope>test</scope>
+      </dependency>
```

Comment:
Let's discuss this - what are these rewrite-java-* libraries? Should they all be included by default, or should they be put into profiles?

Reply:
@JonathanGiles The reason for these test dependencies comes from here:
https://docs.openrewrite.org/authoring-recipes/recipe-development-environment

Essentially, the rewrite-java-"java version" is a test dependency that needs to match your running java version. Because our CI pipelines test on various java versions, we need multiple rewrite-java-"java version" dependencies to account for that.

According to OpenRewrite's documentation, there's no dependency conflict risk from having multiple versions as test dependencies.